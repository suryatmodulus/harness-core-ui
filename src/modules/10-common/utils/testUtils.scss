/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.browser {
  border: 1px solid #cccccc;
  border-radius: 8px;
  min-height: 300px;
  overflow: auto;

  .browser-header {
    background-color: #cccccc;
    padding: 8px 16px;
    display: flex;
  }

  .browser-path {
    display: block;
    border: none;
    line-height: 14px;
    font-size: 12px;
    color: #444444;
    border-radius: 9px;
    min-width: 100px;
    padding: 2px 8px;
    flex: 1 1 auto;
    border: 1px solid transparent;

    &:focus {
      border-color: rgb(30, 167, 253);
      outline: none;
    }
  }

  .browser-content {
    padding: 8px;
  }
}
