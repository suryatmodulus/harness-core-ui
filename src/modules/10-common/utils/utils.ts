/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@harness/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}

export const formatCount = (num: number): string => {
  const min = 1e3
  if (num >= min) {
    const units = ['k', 'M', 'B', 'T']
    const order = Math.floor(Math.log(num) / Math.log(1000))
    const finalNum = Math.round(num / 1000 ** order)
    if (finalNum === min && order < units.length) {
      return 1 + units[order]
    }
    return finalNum + units[order - 1]
  }
  return num.toLocaleString()
}
