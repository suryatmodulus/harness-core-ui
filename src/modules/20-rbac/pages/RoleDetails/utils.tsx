/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
 */

import produce from 'immer'
import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import type { Permission, PermissionResponse } from 'services/rbac'

export const getPermissionMap = (permissionList?: PermissionResponse[]): Map<ResourceType, Permission[]> => {
  const permissionsMap: Map<ResourceType, Permission[]> = new Map()
  permissionList?.map(response => {
    if (response.permission.resourceType) {
      if (permissionsMap.has(response.permission.resourceType as ResourceType)) {
        permissionsMap.get(response.permission.resourceType as ResourceType)?.push(response.permission)
      } else permissionsMap.set(response.permission.resourceType as ResourceType, [response.permission])
    }
  })
  return permissionsMap
}

export const onPermissionChange = (
  permission: string,
  isAdd: boolean,
  permissions: string[],
  setPermissions: (value: React.SetStateAction<string[]>) => void
): void => {
  if (isAdd) {
    if (
      permission === PermissionIdentifier.EDIT_DASHBOARD ||
      permissions.indexOf(PermissionIdentifier.EDIT_DASHBOARD) !== -1
    ) {
      setPermissions(_permissions => [...permissions, permission, PermissionIdentifier.VIEW_DASHBOARD])
    } else {
      setPermissions(_permissions => [...permissions, permission])
    }
  } else if (
    !(
      permission === PermissionIdentifier.VIEW_DASHBOARD &&
      permissions.indexOf(PermissionIdentifier.EDIT_DASHBOARD) !== -1
    )
  ) {
    setPermissions(_permissions =>
      produce(_permissions, draft => {
        draft?.splice(permissions.indexOf(permission), 1)
      })
    )
  }
}
