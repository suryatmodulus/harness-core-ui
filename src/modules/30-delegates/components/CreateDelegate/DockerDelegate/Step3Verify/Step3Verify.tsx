/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useState } from 'react'
import { useParams } from 'react-router-dom'

import { Button, Layout, StepProps } from '@wings-software/uicore'
import { useStrings } from 'framework/strings'
import { useCreateDelegateGroup } from 'services/portal'

import { useToaster } from '@common/exports'
import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import StepProcessing from '../../components/StepProcessing/StepProcessing'
import type { DockerDelegateWizardData } from '../CreateDockerDelegate'

import css from '../Step1Setup/Step1Setup.module.scss'

interface StepSuccessVerificationProps {
  onClose?: any
}

const Step3Verify: React.FC<StepProps<DockerDelegateWizardData> & StepSuccessVerificationProps> = props => {
  const { accountId, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps>()
  const [isVerifiedSuccessfully, setIsVerifiedSuccessfully] = useState(false)
  const { previousStep, prevStepData } = props
  const { getString } = useStrings()
  const { showError } = useToaster()

  const onClickBack = (): void => {
    if (previousStep) {
      previousStep(prevStepData)
    }
  }

  const { mutate: createDelegateGroup } = useCreateDelegateGroup({
    queryParams: {
      accountId
    }
  })

  const onClickDone = async () => {
    if (!isVerifiedSuccessfully) {
      const dockerData = {
        delegateType: 'DOCKER',
        orgIdentifier,
        projectIdentifier,
        name: prevStepData?.name,
        identifier: prevStepData?.identifier,
        description: prevStepData?.description,
        tags: prevStepData?.tags,
        tokenName: prevStepData?.tokenName
      } as any
      const response = (await createDelegateGroup(dockerData)) as any
      if (response?.ok) {
        props?.onClose()
      } else {
        const err = (response as any)?.responseMessages?.[0]?.message
        showError(err)
      }
    } else {
      props?.onClose()
    }
  }

  return (
    <>
      <Layout.Horizontal className={css.baseContainer}>
        <Layout.Vertical className={css.leftPanel}>
          <StepProcessing
            name={props?.prevStepData?.name}
            replicas={1}
            onSuccessHandler={() => setIsVerifiedSuccessfully(true)}
          />
        </Layout.Vertical>
      </Layout.Horizontal>
      <Layout.Horizontal padding="xxxlarge">
        <Button
          id="stepReviewScriptBackButton"
          text={getString('back')}
          onClick={onClickBack}
          icon="chevron-left"
          margin={{ right: 'small' }}
        />
        <Button text={getString('done')} intent="primary" padding="small" onClick={onClickDone} />
      </Layout.Horizontal>
    </>
  )
}

export default Step3Verify
