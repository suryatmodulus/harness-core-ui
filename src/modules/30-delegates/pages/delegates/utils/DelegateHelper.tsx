/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { useStrings } from 'framework/strings'
import { DelegateTypes } from '@delegates/constants'

export const GetDelegateTitleTextByType = (type: string): string => {
  const { getString } = useStrings()

  switch (type) {
    case DelegateTypes.KUBERNETES_CLUSTER:
      return getString('kubernetesText')
    default:
      /* istanbul ignore next */
      return ''
  }
}
