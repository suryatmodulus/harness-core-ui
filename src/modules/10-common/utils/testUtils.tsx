/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { UseGetProps, UseGetReturn, RestfulProvider } from 'restful-react'
import { queryByAttribute } from '@testing-library/react'
import { compile } from 'path-to-regexp'
import { createMemoryHistory } from 'history'
import { Router, Route, Switch, useLocation, useHistory } from 'react-router-dom'
import { ModalProvider } from '@wings-software/uicore'
import qs from 'qs'

import { enableMapSet } from 'immer'
import { AppStoreContext, AppStoreContextProps } from 'framework/AppStore/AppStoreContext'
import { LicenseStoreContext, LicenseStoreContextProps } from 'framework/LicenseStore/LicenseStoreContext'
import { LICENSE_STATE_VALUES } from 'framework/LicenseStore/licenseStoreUtil'
import { withAccountId, accountPathProps } from '@common/utils/routeUtils'
import type { Project } from 'services/cd-ng'
import { StringsContext } from 'framework/strings'

import { FeaturesContext, FeaturesContextProps } from 'framework/featureStore/FeaturesContext'
import type { FeatureDetail, FeatureMetaData } from 'framework/featureStore/featureStoreUtil'
import type { FeatureIdentifier } from 'framework/featureStore/FeatureIdentifier'
import './testUtils.scss'
import { PermissionsContext, PermissionsContextProps } from 'framework/rbac/PermissionsContext'

export type UseGetMockData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Required<
  UseGetProps<TData, TError, TQueryParams, TPathParams>
>['mock']

export interface UseGetMockDataWithMutateAndRefetch<T> extends UseGetMockData<T> {
  mutate: () => Record<string, unknown>
  refetch: () => Record<string, unknown>
  cancel?: () => Record<string, unknown>
}

export interface UseMutateMockData<TData, TRequestBody = unknown> {
  loading?: boolean
  mutate?: (data?: TRequestBody) => Promise<TData>
}

export type UseGetReturnData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Omit<
  UseGetReturn<TData, TError, TQueryParams, TPathParams>,
  'absolutePath' | 'cancel' | 'response'
>

export const findDialogContainer = (): HTMLElement | null => document.querySelector('.bp3-dialog')
export const findPopoverContainer = (): HTMLElement | null => document.querySelector('.bp3-popover-content')

export interface TestWrapperProps {
  path?: string
  pathParams?: Record<string, string | number>
  queryParams?: Record<string, unknown>
  defaultAppStoreValues?: Partial<AppStoreContextProps>
  defaultLicenseStoreValues?: Partial<LicenseStoreContextProps>
  defaultPermissionValues?: Partial<PermissionsContextProps>
  defaultFeaturesValues?: Partial<FeaturesContextProps>
  projects?: Project[]
  enableBrowserView?: boolean
  stringsData?: Record<string, string>
  getString?(key: string): string
}

export const prependAccountPath = (path: string): string => withAccountId(() => path)(accountPathProps)

export const CurrentLocation = (): JSX.Element => {
  const location = useLocation()
  return (
    <div>
      <h1>Not Found</h1>
      <div data-testid="location">{`${location.pathname}${
        location.search ? `?${location.search.replace(/^\?/g, '')}` : ''
      }`}</div>
    </div>
  )
}

export interface BrowserViewProps {
  enable?: boolean
  children: React.ReactNode
}

export function BrowserView(props: BrowserViewProps): React.ReactElement {
  const { enable, children } = props
  const location = useLocation()
  const history = useHistory()

  if (!enable) {
    return <>{children}</>
  }

  function handlePathChange(e: React.ChangeEvent<HTMLInputElement>): void {
    history.replace(e.currentTarget.value)
  }

  const search = location.search ? `?${location.search.replace(/^\?/, '')}` : ''

  return (
    <div className="browser">
      <div className="browser-header">
        <input className="browser-path" value={location.pathname + search} onChange={handlePathChange} />
      </div>
      <div className="browser-content">{children}</div>
    </div>
  )
}

export const TestWrapper: React.FC<TestWrapperProps> = props => {
  enableMapSet()
  const {
    path = '/',
    pathParams = {},
    defaultAppStoreValues,
    queryParams = {},
    defaultLicenseStoreValues,
    defaultPermissionValues,
    defaultFeaturesValues,
    stringsData = {},
    getString = (key: string) => key
  } = props

  const search = qs.stringify(queryParams, { addQueryPrefix: true })
  const routePath = compile(path)(pathParams) + search

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const history = React.useMemo(() => createMemoryHistory({ initialEntries: [routePath] }), [])

  const defaultReturn = {
    enabled: true
  }

  /** TODO: Try fixing this later. This is causing some tests to fail */
  // React.useEffect(() => {
  //   history.replace(compile(path)(pathParams) + qs.stringify(queryParams, { addQueryPrefix: true }))
  //   // eslint-disable-next-line react-hooks/exhaustive-deps
  // }, [path, pathParams, queryParams])

  return (
    <StringsContext.Provider value={{ data: stringsData as any, getString }}>
      <AppStoreContext.Provider
        value={{
          featureFlags: {
            FEATURE_ENFORCEMENT_ENABLED: true
          },
          updateAppStore: () => void 0,
          currentUserInfo: {},
          ...defaultAppStoreValues
        }}
      >
        <LicenseStoreContext.Provider
          value={{
            versionMap: {},
            licenseInformation: {},
            CI_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            FF_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            CCM_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            CD_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            updateLicenseStore: () => void 0,
            ...defaultLicenseStoreValues
          }}
        >
          <PermissionsContext.Provider
            value={{
              permissions: new Map<string, boolean>(),
              requestPermission: () => void 0,
              checkPermission: () => true,
              cancelRequest: () => void 0,
              ...defaultPermissionValues
            }}
          >
            <FeaturesContext.Provider
              value={{
                features: new Map<FeatureIdentifier, FeatureDetail>(),
                featureMap: new Map<FeatureIdentifier, FeatureMetaData>(),
                getEdition: () => {
                  return undefined
                },
                requestFeatures: () => void 0,
                requestLimitFeature: () => void 0,
                checkFeature: () => {
                  return defaultReturn
                },
                checkLimitFeature: () => {
                  return defaultReturn
                },
                getRestrictionType: () => {
                  return undefined
                },
                ...defaultFeaturesValues
              }}
            >
              <Router history={history}>
                <ModalProvider>
                  <RestfulProvider base="/">
                    <BrowserView enable={props.enableBrowserView}>
                      <Switch>
                        <Route exact path={path}>
                          {props.children}
                        </Route>
                        <Route>
                          <CurrentLocation />
                        </Route>
                      </Switch>
                    </BrowserView>
                  </RestfulProvider>
                </ModalProvider>
              </Router>
            </FeaturesContext.Provider>
          </PermissionsContext.Provider>
        </LicenseStoreContext.Provider>
      </AppStoreContext.Provider>
    </StringsContext.Provider>
  )
}

export const queryByNameAttribute = (name: string, container: HTMLElement): HTMLElement | null =>
  queryByAttribute('name', container, name)
