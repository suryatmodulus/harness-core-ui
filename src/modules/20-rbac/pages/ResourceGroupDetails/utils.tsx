import produce from 'immer'
import { RbacResourceGroupTypes } from '@rbac/constants/utils'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import { isDynamicResourceSelector } from '@rbac/utils/utils'
import type { ResourceSelector } from 'services/resourcegroups'

export const getSelectedResourcesMap = (
  resourceSelectorList?: ResourceSelector[]
): Map<ResourceType, string[] | string> => {
  const map: Map<ResourceType, string[] | string> = new Map()

  resourceSelectorList?.map(resourceSelector => {
    if (isDynamicResourceSelector(resourceSelector.type)) {
      map.set(resourceSelector.resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
    } else {
      map.set(resourceSelector.resourceType, resourceSelector.identifiers)
    }
  })
  return map
}

export const validateResourceSelectors = (value: ResourceSelector[]): ResourceType[] => {
  return value
    .filter(val => {
      if (!isDynamicResourceSelector(val.type)) {
        if (val.identifiers?.length < 1) {
          return true
        }
      }
      return false
    })
    .map(val => val.resourceType)
}

export const getResourceSelectorsfromMap = (map: Map<ResourceType, string[] | string>): ResourceSelector[] => {
  const resourceSelectors: ResourceSelector[] = []
  map.forEach((value, key) => {
    if (isDynamicResourceSelector(value)) {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR,
        resourceType: key
      })
    } else {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.STATIC_RESOURCE_SELECTOR,
        resourceType: key,
        identifiers: value
      })
    }
  })
  return resourceSelectors
}

export const computeResourceMapOnChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  resourceType: ResourceType,
  isAdd: boolean,
  identifiers?: string[]
): void => {
  if (identifiers) {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, identifiers)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          if (Array.isArray(draft.get(resourceType))) {
            draft.set(
              resourceType,
              (draft.get(resourceType) as string[]).filter(el => !identifiers.includes(el))
            )
          }
          if (draft.get(resourceType)?.length === 0) {
            draft.delete(resourceType)
          }
        })
      )
    }
  } else {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.delete(resourceType)
        })
      )
    }
  }
}

export const computeResourceMapOnMultiChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  types: ResourceType[],
  isAdd: boolean
): void => {
  if (isAdd) {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      })
    )
  } else {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.delete(resourceType)
        })
      })
    )
  }
}
