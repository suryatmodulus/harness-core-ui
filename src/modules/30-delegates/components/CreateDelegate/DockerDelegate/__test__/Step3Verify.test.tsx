/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { TestWrapper } from '@common/utils/testUtils'
import Step3Verify from '../Step3Verify/Step3Verify'

const createGroupFn = jest.fn().mockReturnValue({
  ok: true
})
jest.mock('services/portal', () => ({
  useCreateDelegateGroup: jest.fn().mockImplementation(() => {
    return {
      mutate: createGroupFn
    }
  }),
  useGetDelegatesHeartbeatDetailsV2: jest.fn().mockImplementation(() => {
    return { data: {}, refetch: jest.fn(), error: null, loading: false }
  })
}))

describe('Create Docker Step3Verify', () => {
  test('render data and go back', () => {
    const { container } = render(
      <TestWrapper>
        <Step3Verify />
      </TestWrapper>
    )

    const backBtn = container.querySelector('button') as HTMLButtonElement

    userEvent.click(backBtn!)

    expect(container).toMatchSnapshot()
  })
})
