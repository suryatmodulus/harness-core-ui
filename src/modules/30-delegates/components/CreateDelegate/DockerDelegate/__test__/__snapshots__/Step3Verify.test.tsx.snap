// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Docker Step3Verify render data and go back 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main baseContainer"
  >
    <div
      class="Layout--vertical StyledProps--main"
    >
      <div
        class="Layout--vertical Layout--layout-spacing-xxlarge StyledProps--main"
      >
        <div
          class="Layout--horizontal StyledProps--main StyledProps--padding-large"
        >
          <span
            class="bp3-icon StyledProps--main"
            data-icon="steps-spinner"
            style="margin-right: 12px;"
          >
            <svg
              height="16"
              viewBox="0 0 128 128"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M122.5 69.25H96.47a33.1 33.1 0 000-10.5h26.05a5.25 5.25 0 010 10.5z"
                  fill="#ff8e02"
                />
                <path
                  d="M112.04 97.83L89.47 84.8a33.1 33.1 0 005.25-9.1l22.57 13.03a5.25 5.25 0 01-5.28 9.1zM88.68 117.35L75.65 94.78a33.1 33.1 0 009.1-5.25l13.02 22.57a5.25 5.25 0 11-9.1 5.25zM58.7 122.57V96.5a33.1 33.1 0 0010.5 0v26.07a5.25 5.25 0 01-10.5 0zM30.1 112.1l13.04-22.57a33.1 33.1 0 009.1 5.25L39.2 117.35a5.25 5.25 0 11-9.1-5.25zM10.6 88.74L33.16 75.7a33.1 33.1 0 005.25 9.1L15.88 97.83a5.25 5.25 0 11-5.25-9.1z"
                  fill="#ffddb3"
                  fill-opacity="0.3"
                />
                <path
                  d="M5.37 58.75h26.06a33.1 33.1 0 000 10.5H5.37a5.25 5.25 0 010-10.5z"
                  fill="#ffd29a"
                  fill-opacity="0.4"
                />
                <path
                  d="M15.85 30.17L38.4 43.2a33.1 33.1 0 00-5.24 9.1L10.6 39.25a5.25 5.25 0 115.25-9.1z"
                  fill="#ffc680"
                  fill-opacity="0.5"
                />
                <path
                  d="M39.2 10.65l13.03 22.57a33.1 33.1 0 00-9.1 5.25l-13-22.57a5.25 5.25 0 119.1-5.25z"
                  fill="#ffbb67"
                  fill-opacity="0.6"
                />
                <path
                  d="M69.2 5.43V31.5a33.1 33.1 0 00-10.5 0V5.42a5.25 5.25 0 1110.5 0z"
                  fill="#ffb04d"
                  fill-opacity="0.7"
                />
                <path
                  d="M97.77 15.9L84.75 38.47a33.1 33.1 0 00-9.1-5.25l13.03-22.57a5.25 5.25 0 119.1 5.25z"
                  fill="#ffa535"
                  fill-opacity="0.8"
                />
                <path
                  d="M117.3 39.26L94.7 52.3a33.1 33.1 0 00-5.25-9.1l22.57-13.03a5.25 5.25 0 015.25 9.1z"
                  fill="#ff991b"
                  fill-opacity="0.9"
                />
                <animatetransform
                  attributeName="transform"
                  calcMode="discrete"
                  dur="1200ms"
                  repeatCount="indefinite"
                  type="rotate"
                  values="0 64 64;30 64 64;60 64 64;90 64 64;120 64 64;150 64 64;180 64 64;210 64 64;240 64 64;270 64 64;300 64 64;330 64 64"
                />
              </g>
            </svg>
          </span>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-small"
          >
            delegate.successVerification.checkDelegateInstalled
          </p>
        </div>
        <div
          class="Layout--vertical StyledProps--main StyledProps--padding-large"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main checkItemsWrapper"
          >
            <span
              class="bp3-icon StyledProps--main checkIcon StyledProps--color StyledProps--color-yellow500"
              data-icon="command-artifact-check"
            >
              <svg
                height="10"
                viewBox="0 0 14 14"
                width="10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 14c-3.9 0-7-3.1-7-7s3.1-7 7-7 7 3.1 7 7-3.1 7-7 7zm3.3-10c-.2 0-.3.1-.4.2L5.7 8.4 4.1 6.9c-.1-.2-.2-.2-.4-.2-.4 0-.7.3-.7.6 0 .2.1.4.2.5l2 2c.1.1.3.2.5.2s.3-.1.4-.2l4.7-4.7c.1-.1.2-.2.2-.4 0-.4-.3-.7-.7-.7z"
                  fill="#4d5969"
                />
              </svg>
            </span>
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-weight-bold"
            >
              delegate.successVerification.heartbeatReceived
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Layout--horizontal StyledProps--main StyledProps--padding-xxxlarge"
  >
    <button
      class="bp3-button bp3-disabled bp3-loading Button--button StyledProps--font StyledProps--main StyledProps--margin-right-small Button--with-current-color Button--withLeftIcon"
      disabled=""
      id="stepReviewScriptBackButton"
      tabindex="-1"
      type="button"
    >
      <div
        class="bp3-spinner bp3-button-spinner"
      >
        <div
          class="bp3-spinner-animation"
        >
          <svg
            height="20"
            stroke-width="16.00"
            viewBox="-3.00 -3.00 106.00 106.00"
            width="20"
          >
            <path
              class="bp3-spinner-track"
              d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
            />
            <path
              class="bp3-spinner-head"
              d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
              pathLength="280"
              stroke-dasharray="280 280"
              stroke-dashoffset="210"
            />
          </svg>
        </div>
      </div>
      <span
        class="bp3-icon bp3-icon-chevron-left StyledProps--main StyledProps--padding-right-xsmall"
        icon="chevron-left"
      >
        <svg
          data-icon="chevron-left"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            chevron-left
          </desc>
          <path
            d="M7.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71l-4 4C5.11 7.47 5 7.72 5 8c0 .28.11.53.29.71l4 4a1.003 1.003 0 001.42-1.42L7.41 8z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        class="bp3-button-text"
      >
        back
      </span>
    </button>
    <button
      class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--padding-small Button--with-current-color"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        done
      </span>
    </button>
  </div>
</div>
`;
